pyhap/characteristic.py

```diff
@@ -91,60 +89,49 @@ def __init__(self, display_name, type_id, properties, value=None, broker=None):
 
         :param properties: A dict of properties, such as Format, ValidValues, etc.
         :type properties: dict
-
-        :param value: The initial value to set to this characteristic. If no value is given,
-            the assigned value happens as:
-            - if there is a ValidValue property, use some value from it.
-            - else use `HAP_FORMAT.DEFAULT` for the format of this characteristic.
-        :type value: Depends on `properties["Format"]`
         """
-        assert "Format" in properties and "Permissions" in properties
         self.display_name = display_name
         self.type_id = type_id
         self.properties = properties
-        if value:
-            self.value = value
-        else:
-            if self.properties.get('ValidValues'):
-                self.value = min(self.properties['ValidValues'].values())
-            else:
-                self.value = HAP_FORMAT.DEFAULT[properties["Format"]]
-        self.broker = broker
+        self.broker = None
         self.setter_callback = None
+        self.value = self._get_default_value()
 
     def __repr__(self):
         """Return the representation of the characteristic."""
-        return "<characteristic display_name='{}' value={} properties={}>" \
+        return '<characteristic display_name={} value={} properties={}>' \
             .format(self.display_name, self.value, self.properties)
 
-    def set_value(self, value, should_notify=True, should_callback=True):
-        """Set the given raw value. It is checked if it is a valid value.
-
-        :param value: The value to assign as this Characteristic's value.
-        :type value: Depends on properties["Format"]
+    def set_value(self, value):
```

Comment:
I am still thinking about the `should_notify` and if there is a case for it. Image you want to set the value but not notify iOS clients - technically you can do it by setting the value property but then you don't have the `to_valid_value` transformation. What do you think, is it worth adding it?

Reply:
I though about this case as well. It's a trade of here. I think that e.g. `value = 0` would be enough, since most of those cases are for static content which is hard coded. Therefore it would be quite easy to recognize a problem / bug and change it.

Reply:
Yeah adding a `should_notify=True` would probably be a good move.

Reply:
I guess it wont hurt anyone with `should_notify=True`